homeassistant/helpers/dispatcher.py

```diff
@@ -236,4 +243,13 @@ def async_dispatcher_send_internal[*_Ts](
         if job is None:
             job = _generate_job(signal, target)
             target_list[target] = job
-        hass.async_run_hass_job(job, *args)
+        # We do not wrap Callback jobs in catch_log_exception since
+        # single use dispatchers spend more time wrapping the callback
+        # than the actual callback takes to run in many cases.
+        if job.job_type is HassJobType.Callback:
```

Comment:
The irony here is this check is effectively the same as `async_run_hass_job` does so it would be nice to bypass and call `hass._async_add_hass_job` but thats protected.  In practice it doesn't likely matter since almost all dispatcher calls are going to be callbacks anyways